src/ParallelRun.jl

```diff
@@ -296,8 +299,388 @@ function load_images(rcfs, stagedir)
+            cat_local = vcat(entry, neighbors)
+            ids_local = vcat(entry_id, neighbor_ids)
+            
+            #vp = Vector{Float64}[init_source(ce) for ce in cat_local]
+            vp = Vector{Float64}[haskey(target_source_variational_params, x) ? target_source_variational_params[x] : init_source(catalog[x]) for x in ids_local]
+            patches, tile_source_map = Infer.get_tile_source_map(images, cat_local)
+            ea = ElboArgs(images, vp, tile_source_map, patches, [1])
+            Infer.fit_object_psfs!(ea, ea.active_sources)
+            Infer.load_active_pixels!(ea)
+            @assert length(ea.active_pixels) > 0
+            model[cur_source_index] = ea
+        end
+    end
+
+    # Initialize elboargs in parallel
+    tic()
+    thread_initialize_sources_assignment = partition_equally(nprocthreads, n_sources)
+    
+    Threads.@threads for i=1:nprocthreads
+        for batch = 1:length(thread_initialize_sources_assignment[i])
+            initialize_elboargs_sources(thread_initialize_sources_assignment[i][batch])
+        end
+    end
+
+    Log.info("Done preallocating array of elboargs. Elapsed time: $(toq())")
+
+    # Keep track of object values achieved per light source
+    obj_values = Array{Float64}(length(target_sources))
+
+    # Process partition of sources. Multiple threads call this function in parallel.
+    function process_sources(source_assignment::Vector{Int64})
+        for cur_source_indx in source_assignment
+            cur_entry = catalog[target_sources[cur_source_indx]]
+            nputs(dt_nodeid, "Thread $(Threads.threadid()) processing source $(target_sources[cur_source_indx]): objid = $(cur_entry.objid)")
+            nputs(dt_nodeid, "Before: $(model[cur_source_indx].vp[1])")
+            iter_count, obj_value, max_x, r = DeterministicVI.maximize_f(DeterministicVI.elbo, model[cur_source_indx], max_iters=10)
+            nputs(dt_nodeid, "After: $(model[cur_source_indx].vp[1])")
+            obj_values[cur_source_indx] = obj_value
+        end
+    end
+
+
+    # Process sources in parallel using nprocthreads.
+    tic()
+    n_batches = length(thread_sources_assignment[1])
+    for iter = 1:n_iters
+        # Process every batch of every iteration. We do the batches on the outside
+        # Since there is an implicit barrier after the inner threaded for loop below.
+        # We want this barrier because there may be conflict _between_ Cyclades batches.
+        for batch = 1:n_batches
+            # Process every batch of every iteration with nprocthreads
+            Threads.@threads for i = 1:nprocthreads
+                process_sources(thread_sources_assignment[i][batch])
+            end
+        end
+    end
+    Log.info("Done fitting elboargs. Elapsed time: $(toq())")    
+    
+    # Return add results to dictionary
+    # TODO max: parallelize?
```

Comment:
I see, it seems like a vector of InferResult is used instead of a map. Should I wait until that branch is merged with master to change it?